assets/src/components/v2/elevator/elevator_closures.tsx

```diff
@@ -1,211 +1,76 @@
-  }, [stations]);
-
-  const getPagingIndicators = (num: number) => {
-    const indicators: JSX.Element[] = [];
-    for (let i = 0; i < num; i++) {
-      const indicator =
-        pageIndex === i ? (
-          <PagingDotSelected
-            className="paging-indicator"
-            height={40}
-            width={40}
-            key={i}
-          />
-        ) : (
-          <PagingDotUnselected
-            className="paging-indicator"
-            height={28}
-            width={28}
-            key={i}
-          />
-        );
-      indicators.push(indicator);
-    }
-
-    return indicators;
-  };
-
-  return (
-    <div className="outside-closure-list">
-      <div className="header-container">
-        <div className="header">
-          <div className="header__title">MBTA Elevator Closures</div>
-          <div>
-            <AccessibilityAlert height={128} width={155} />
-          </div>
-        </div>
-        <hr className="thin" />
-      </div>
-      <div className="closure-list-container">
-        {
-          <div
-            className="closure-list"
-            style={
-              {
-                "--closure-list-offset": pageIndex,
-              } as React.CSSProperties
-            }
-            ref={ref}
-          >
-            {stations.map((station) => (
-              <ClosureRow station={station} key={station.id} />
-            ))}
-          </div>
-        }
-      </div>
-      {numPages > 1 && (
-        <div className="paging-info-container">
-          <div className="more-elevators-text">
-            +{numOffsetRows} more elevators
-          </div>
-          <div className="paging-indicators">
-            {getPagingIndicators(numPages)}
-          </div>
-        </div>
-      )}
-    </div>
-  );
-};
-
 interface Props extends WrappedComponentProps {
   id: string;
   in_station_closures: ElevatorClosure[];
   other_stations_with_closures: StationWithClosures[];
+  alternate_direction_text: string;
+  accessible_path_direction_arrow: Direction;
+  accessible_path_image_url: string | null;
+  accessible_path_image_here_coordinates: Coordinates;
 }
 
 const ElevatorClosures: React.ComponentType<Props> = ({
+  id,
   other_stations_with_closures: otherStationsWithClosures,
   in_station_closures: inStationClosures,
+  alternate_direction_text: alternateDirectionText,
+  accessible_path_direction_arrow: accessiblePathDirectionArrow,
+  accessible_path_image_url: accessiblePathImageUrl,
+  accessible_path_image_here_coordinates: accessiblePathImageHereCoordinates,
   lastUpdate,
   onFinish,
 }: Props) => {
+  const currentElevatorClosure = inStationClosures.find(
+    (c) => c.elevator_id === id,
+  );
+
   return (
     <div className="elevator-closures">
-      <InStationSummary closures={inStationClosures} />
-      <OutsideClosureList
-        stations={otherStationsWithClosures}
-        lastUpdate={lastUpdate}
-        onFinish={onFinish}
-      />
+      {currentElevatorClosure ? (
```

Comment:
We might have already talked about this, but I think it could be worthwhile to move this conditional to the widget generation. This single "widget" is basically two widgets in a trenchcoat — we always send over all the fields for both, and then throw half of them away, depending on a condition the server itself could easily check. In other words, we could instead have a `CurrentElevatorClosed` widget and an `OutsideClosures` widget on the backend, each populated with only the fields needed for that "view".

I could be missing something about future plans for this widget that might require it to have access to both sets of information at the same time, though.

Reply:
Big fan of that approach. That would likely be a pretty big diff though. You think it makes sense here or should I create a followup PR?

Reply:
I think within this PR is actually the best place for it, since this is where we're introducing the second widget. Else we'd have (in the final commits on `main`) one commit that adds the widget using this approach, and a second that completely changes the approach. That's a lot of extra, and ultimately pointless, information for anyone looking at history to sort through.

Reply:
Should be good to go if you want to take a look.